src/main/java/dan200/computercraft/client/render/TileEntityMonitorRenderer.java

```diff
@@ -158,4 +142,92 @@ public void render( @Nonnull TileMonitor monitor, float partialTicks, @Nonnull M
+                        IDENTITY, builder, 0, 0,
+                        terminal, !monitor.isColour(), yMargin, yMargin, xMargin, xMargin
+                    );
+
+                    builder.finishDrawing();
+                    vbo.upload( builder );
+                }
+
+                vbo.bindBuffer();
+                FixedWidthFontRenderer.TYPE.getVertexFormat().setupBufferState( 0L );
+                vbo.draw( matrix, FixedWidthFontRenderer.TYPE.getDrawMode() );
+                VertexBuffer.unbindBuffer();
+                FixedWidthFontRenderer.TYPE.getVertexFormat().clearBufferState();
+                break;
+            }
+
+            case TBO:
+            {
+                // TODO: Test whether we need this, or if 0 is enough. I'm curious as to how this plays with Optifine's
+                //  shaders.
+                int program = GlStateManager.getInteger( GL20.GL_CURRENT_PROGRAM );
+
+                if( !MonitorTextureBufferShader.use() ) return;
+
+                int width = terminal.getWidth(), height = terminal.getHeight();
+                int pixelWidth = width * FONT_WIDTH, pixelHeight = height * FONT_HEIGHT;
+
+                if( redraw )
+                {
+                    ByteBuffer buffer = GLAllocation.createDirectByteBuffer( width * height * 3 );
+                    for( int y = 0; y < height; y++ )
+                    {
+                        TextBuffer text = terminal.getLine( y ), textColour = terminal.getTextColourLine( y ), background = terminal.getBackgroundColourLine( y );
+                        for( int x = 0; x < width; x++ )
+                        {
+                            buffer.put( (byte) (text.charAt( x ) & 0xFF) );
+                            buffer.put( (byte) getColour( textColour.charAt( x ), Colour.WHITE ) );
+                            buffer.put( (byte) getColour( background.charAt( x ), Colour.BLACK ) );
+                        }
+                    }
+                    buffer.flip();
+
+                    GlStateManager.bindBuffer( GL31.GL_TEXTURE_BUFFER, monitor.tboBuffer );
+                    GlStateManager.bufferData( GL31.GL_TEXTURE_BUFFER, buffer, GL20.GL_STATIC_DRAW );
+                    GlStateManager.bindBuffer( GL31.GL_TEXTURE_BUFFER, 0 );
+                }
+
+                // Nobody knows what they're doing!
+                GlStateManager.activeTexture( MonitorTextureBufferShader.TEXTURE_INDEX );
+                GL11.glBindTexture( GL31.GL_TEXTURE_BUFFER, monitor.tboTexture );
+                GlStateManager.activeTexture( GL13.GL_TEXTURE0 );
+
+                MonitorTextureBufferShader.setupUniform( matrix, width, height, terminal.getPalette(), !monitor.isColour() );
+
+                Tessellator tessellator = Tessellator.getInstance();
+                BufferBuilder buffer = tessellator.getBuffer();
+                buffer.begin( GL11.GL_QUADS, DefaultVertexFormats.POSITION );
```

Comment:
This should be converted to a `GL_TRIANGLE_STRIP`. `GL_QUADS` has been deprecated, and has a slight performance drawback, due to the CPU having to triangulate the mesh before rendering.